src/bindgen/parser.rs

```diff
@@ -563,6 +572,71 @@ impl Parse {
         nested_modules
     }
 
+    fn load_syn_impl(&mut self, item_impl: &syn::ItemImpl){
+
+        // Check for trait impl
+        let trait_path = if let Some((_, path, _)) = &item_impl.trait_{ path } else { return };
+
+        for impl_item in &item_impl.items{
+
+            // Check for associated type
+            if let syn::ImplItem::Type(impl_item_type) = impl_item{
+
+                // Parse trait path
+                let trait_ = trait_path.segments.last().unwrap().ident.to_string();
+                let trait_ = Path::new(trait_);
+                
+                // Parse type
+                let impl_type = &item_impl.self_ty;
+                let ty = match Type::load(impl_type){
+                    Ok(opt) => {
+                        match opt{
+                            Some(output) => output,
+                            None => {
+                                warn!("Err parsing assoc type in impl: Valid but empty type");
+                                continue;
+                            } 
+                        }
+                    },
+                    Err(msg) => {
+                        warn!("Err parsing assoc type in impl: {}", msg);
+                        continue;
+                    }
+                };
+
+                let ident = impl_item_type.ident.to_string();
+                let ident = Path::new(ident);
+                
+                let key = AssocTypeId{
+                    ty: Box::new(ty),
+                    trait_,
+                    ident,
+                };
+
+                // Parse concrete type
+                let syn_type = &impl_item_type.ty;
+                let conc_type = match Type::load(syn_type){
+                    Ok(opt) => {
+                        match opt{
+                            Some(output) => output,
+                            None => {
+                                warn!("Err parsing assoc type in impl: Valid but empty type");
+                                continue;
+                            } 
+                        }
+                    },
+                    Err(msg) => {
+                        warn!("Err parsing assoc type in impl: {}", msg);
+                        continue;
+                    }
+                };
```

Comment:
this is duplicated, please put into a function

Reply:
I don't understand this. Which parts should I put into a function. This part was weird because Type::load returns a Result<Option<Type>, String>